hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -422,7 +422,10 @@ private void handleUserTransaction(
 
                     // Dispatch the transaction to the handler
                     dispatcher.dispatchHandle(context);
-                    recordBuilder.status(SUCCESS);
+                    // Do not override to SUCCESS if there is an error status
+                    if (recordBuilder.status() == OK) {
+                        recordBuilder.status(SUCCESS);
```

Comment:
@tinker-michaelj @Neeharika-Sompalli 

Is this check okay? In the current flow, we were setting record builder status [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java#L87) in EthereumTransactionHandler, and no matter what it is, we always override it to `SUCCESS` in HandlerWorkflow. That's not correct, right?

Reply:
This check looks suspicious. We should never reach this point with a status other than `OK`. If there was an error, we have to throw an Exception. Otherwise, the error handling will not work (e.g., the state changes are not reverted).